File: db4o.net/Db4objects.Db4o.Tests/Db4objects.Db4o.Tests/Common/Concurrency/ParameterizedEvaluationTestCase.cs
/* Copyright (C) 2004 - 2011  Versant Inc.  http://www.db4o.com */

#if !SILVERLIGHT
using Db4oUnit;
using Db4oUnit.Extensions;
using Db4objects.Db4o;
using Db4objects.Db4o.Ext;
using Db4objects.Db4o.Query;
using Db4objects.Db4o.Tests.Common.Concurrency;

namespace Db4objects.Db4o.Tests.Common.Concurrency
{
	public class ParameterizedEvaluationTestCase : Db4oClientServerTestCase
	{
		public static void Main(string[] args)
		{
			new ParameterizedEvaluationTestCase().RunConcurrency();
		}

		public string str;

		protected override void Store()
		{
			Store("one");
			Store("fun");
			Store("ton");
			Store("sun");
		}

		private void Store(string str)
		{
			ParameterizedEvaluationTestCase pe = new ParameterizedEvaluationTestCase();
			pe.str = str;
			Store(pe);
		}

		public virtual void Conc(IExtObjectContainer oc)
		{
			Assert.AreEqual(2, QueryContains(oc, "un").Count);
		}

		private IObjectSet QueryContains(IExtObjectContainer oc, string str)
		{
			IQuery q = oc.Query();
			q.Constrain(typeof(ParameterizedEvaluationTestCase));
			q.Constrain(new ParameterizedEvaluationTestCase.MyEvaluation(str));
			return q.Execute();
		}

		[System.Serializable]
		public class MyEvaluation : IEvaluation
		{
			public string str;

			public MyEvaluation(string str)
			{
				this.str = str;
			}

			public virtual void Evaluate(ICandidate candidate)
			{
				ParameterizedEvaluationTestCase pe = (ParameterizedEvaluationTestCase)candidate.GetObject
					();
				bool inc = pe.str.IndexOf(str) != -1;
				candidate.Include(inc);
			}
		}
	}
}
#endif // !SILVERLIGHT

File: db4oj.tests/src/com/db4o/db4ounit/common/concurrency/ParameterizedEvaluationTestCase.java
/* Copyright (C) 2004 - 2007  Versant Inc.   http://www.db4o.com */

package com.db4o.db4ounit.common.concurrency;

import com.db4o.*;
import com.db4o.ext.*;
import com.db4o.query.*;

import db4ounit.*;
import db4ounit.extensions.*;

public class ParameterizedEvaluationTestCase extends Db4oClientServerTestCase {
	
	public static void main(String[] args) {
		new ParameterizedEvaluationTestCase().runConcurrency();
	}

	public String str;

	protected void store() {
		store("one");
		store("fun");
		store("ton");
		store("sun");
	}

	private void store(String str) {
		ParameterizedEvaluationTestCase pe = new ParameterizedEvaluationTestCase();
		pe.str = str;
		store(pe);
	}

	public void conc(ExtObjectContainer oc) {
		Assert.areEqual(2, queryContains(oc, "un").size());
	}

	private ObjectSet queryContains(ExtObjectContainer oc, final String str) {
		Query q = oc.query();
		q.constrain(ParameterizedEvaluationTestCase.class);
		q.constrain(new MyEvaluation(str));
		return q.execute();
	}
	
	public static class MyEvaluation implements Evaluation {
		public String str;
		public MyEvaluation(String str) {
			this.str = str;
		}
		public void evaluate(Candidate candidate) {
			ParameterizedEvaluationTestCase pe = (ParameterizedEvaluationTestCase) candidate
					.getObject();
			boolean inc = pe.str.indexOf(str) != -1;
			candidate.include(inc);
		}
	}

}
